src/DistributedLocking/Consul/LockProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Cascade\DistributedLocking\Consul;
+
+use Cascade\DistributedLocking\ILockProvider;
+use Cascade\DistributedLocking\ILockSessionProvider;
+use SensioLabs\Consul\Services\KV;
+
+class LockProvider implements ILockProvider
+{
+    /** @var KV */
+    private $consulKeyValue;
+
+    /** @var ILockSessionProvider */
+    private $sessionProvider;
+
+    /**
+     * @param ILockSessionProvider $sessionProvider
+     * @param KV $consulKeyValue
+     */
+    public function __construct(ILockSessionProvider $sessionProvider, KV $consulKeyValue)
```

Comment:
I need to think about the LockSessionProvider issue -- in theory you could provide anything that implemented ILockSessionProvider and that maintained the session in Consul, so it doesn't have to be the standard Consul\LockSessionProvider -- not sure that would ever come up in practice, but I don't know if there's a reason to prevent it either.